universal-application-tool-0.0.1/app/services/ErrorAnd.java

```diff
@@ -0,0 +1,57 @@
+package services;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+import com.google.common.collect.ImmutableSet;
+
+/**
+ * Wraps two values, labeled "errors" and "result". Errors is intended to be an immutable collection
+ * of error objects while result is the result of an action. If the action was successful only
+ * result need be provided.
+ */
+class ErrorAnd<T, E> {
+  private final ImmutableSet<E> errors;
+  private final T result;
+
+  /** Constructor for the error case. */
+  public ErrorAnd(ImmutableSet<E> errors) {
+    this.errors = checkNotNull(errors);
+    this.result = null;
+  }
+
+  /** Constructor for the error case but when result is also useful. */
+  public ErrorAnd(ImmutableSet<E> errors, T result) {
+    this.errors = checkNotNull(errors);
+    this.result = checkNotNull(result);
+  }
+
+  /** Constructor for the success case. */
+  public ErrorAnd(T result) {
+    this.result = checkNotNull(result);
+    this.errors = ImmutableSet.of();
+  }
+
+  /** True if there is a result */
```

Comment:
nitpick (non-blocking): For consistency with the other docstrings, consider rewording to "Returns true if..."

Reply:
done